sysdeps/lemon/generic/lemon.cpp

```diff
@@ -0,0 +1,50 @@
+#include <lemon/syscall.h>
+#include <stddef.h>
+#include <bits/ensure.h>
+#include <bits/posix/pid_t.h>
+
+namespace mlibc{
+
+	void sys_exit(int status){
+		syscall(SYS_EXIT, status, 0, 0, 0, 0);
+	}
+
+	int sys_anon_allocate(size_t size, void **pointer) {
+		// Make sure to only allocate whole pages
+		__ensure(!(size & 0xFFF));
+		syscall(SYS_ALLOC, (((size / 0x1000)*0x1000 < size) ? (size / 0x1000) + 1 : (size / 0x1000)), (uintptr_t)pointer, 0, 0, 0);
```

Comment:
Rounding up to page size can be simplified: `(size + 0xFFF) & ~static_cast<size_t>(0xFFF)`. This is also faster because it doesn't generate a `div` instruction.